.github/license/license-header.py

```diff
@@ -0,0 +1,237 @@
+
+        if not matched:
+            continue
+
+        content = file_read(filepath)
+        wrap = get_wrapper(filename)
+
+        header_comment = wrap.wrapper(license_header, args)
+
+        start = 0
+        end = 0
+
+        # Look for an exact substr match
+        found = content.find(header_comment, 0, len(header_comment) + args.extra)
+        if found >= 0:
+            if not args.remove:
+                continue
+
+            start = found
+            end = found + len(header_comment)
+        else:
+            # Look for a fuzzy match in the first 60 chars
+            found = regex.search(
+                "(?be)(%s){e<=%d}" % (regex.escape(header_comment[0:60]), 6),
+                content[0 : 80 + args.extra],
+            )
+            if found:
+                fuzzy = regex.compile(
+                    "(?be)(%s){e<=%d}" % (regex.escape(header_comment), args.editdist)
+                )
+
+                # If the first 80 chars match - try harder for the rest of the header
+                found = fuzzy.search(content[0 : len(header_comment) + args.extra], found.start())
+                if found:
+                    start = found.start()
+                    end = found.end()
+
+        if args.remove:
+            if start == 0 and end == 0:
+                if not args.inplace:
+                    print(content, end="")
+
+                continue
+
+            # If removing the header text, zero it out there.
+            header_comment = ""
+
+        if args.check:
+            fail = True
+            continue
+
+        # Remove any partially matching header
+        content = content[0:start] + content[end:]
+
+        if wrap.hashbang:
+            search = regex.search("^#!.*\n", content)
+            if search:
+                content = (
+                    content[search.start() : search.end()]
+                    + header_comment
+                    + content[search.end() :]
+                )
+            else:
+                content = header_comment + content
+        else:
+            content = header_comment + content
+
+        if args.inplace:
+            with open(filepath, "w") as file:
+                print(content, file=file, end="")
+
+        if content != "":
+            print(filepath, end="\n")
+
+        assert content == ""
+
+
+def process_license_header(files, args):
+    llm_on_ray_header_files_globs = []
+    if args.llm_on_ray_header_files:
+        with open(args.llm_on_ray_header_files) as f:
+            llm_on_ray_header_files_globs.extend(line.strip() for line in f)
+
+    llm_on_ray_header_files = []
+    for file in files:
+        if any([fnmatch.fnmatch(file, glob) for glob in llm_on_ray_header_files_globs]):
+            llm_on_ray_header_files.append(file)
+
+    license_header_llm_on_ray = file_lines(args.header)
+    check_license_header(llm_on_ray_header_files, license_header_llm_on_ray, args)
+
+
+fail = False
+
+
+def main():
+    global fail
```

Comment:
It's OK to raise exception from where the error happened instead rather than using global variable to mark a failure.

Reply:
Variable `fail` is used to check whether it is passed and distinguish the return code.
Maybe we can keep it :)